### PR TITLE
docker: Experimentally enable local arm64 or multiarch image builds.

### PR DESCRIPTION
**Pull request by klardotsh**
Allow building amd64+arm64 multiarch images locally for those brave enough to experiment with them.

Refs #357.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,15 +10,20 @@ This is a container image for running [Zulip](https://zulip.com)
...
 
-Project status: **Alpha**. While this project works and is
-used by many sites in production, configuring is substantially more
-error-prone than the [normal Zulip installer][normal-install] (which
-Just Works). We recommend this project if you want to host Zulip
-using Docker, but both setting up and maintaining a Zulip server is
-simpler and less error-prone with the normal installer than with Docker.
+<!-- Remove when https://github.com/zulip/docker-zulip/issues/357 resolved -->
+> `arm64` support is experimental, and is not provided in the Docker Hub
+> images. To build an `arm64` image yourself, see `make help` locally.
```
Comment (andersk) [inline README.md:17]: I would expect you don’t need any of this to build an arm64 image yourself. As long as you’re on an arm64 host, the usual `docker compose build` should build the image automatically. The only advantage of being on x86_64 is that you can pull our pre-built image from Docker Hub with `docker compose pull`.

Comment (klardotsh) [reply README.md:17]: I guess the real selling point here is the cross-compilation: from an x86_64 machine (be that a dev laptop, or CI infrastructure), this allows building arm64 images for deploy to Graviton2, Raspberry Pi, whatever one's infra is. This is a stepping stone in my own eventual hopes for this repo: to fully automate our Docker Hub publishing when new Zulip Server releases are tagged. GitHub Actions is, last I saw, still fully on x86_64 and there aren't imminent plans to change that, so this x-compile tooling will be necessary for us to provide multiarch manifests for folks on such infrastructure to pull.

The linked #357 explicitly asks for us to start supporting ARM64 images on Docker Hub, which I think is a perfectly reasonable request given the spread of that hardware in the wild now.

Comment (andersk) [reply README.md:17]: Absolutely. My comment is just about this specific documentation line.

Comment (klardotsh) [reply README.md:17]: Aha - would something along these lines read more clearly to that end?

> `arm64` support is experimental, and is not provided in Docker Hub yet. On an arm64 host, the image can be build with the standard `docker build` flows; to cross-compile from an amd64 host, see `make help` locally.

Comment (alexmv) [reply README.md:17]: 👍🏻 on that wording from me.

**Review by alexmv — COMMENTED** [review]
I'm not sold on introducing Makefile as a new language here -- I don't think we're gaining anything over a shell script besides arcane syntax.  Specifically, none of the "use intermediate build products" feature which make Makefiles save cycles are in use here.

File: README.md (hunk 1 above)
Comment (klardotsh) [reply README.md:17]: This conversation superceded by the move to GHCR and fully-automated image builds.

**Comment by klardotsh** [comment]
Closing to reopen with full GHA+GHCR automation (which I realize I did on a different branch...)

**Pull request closed without merging**